### ISSUE TITLE
Batch Normalization Patch

### ISSUE THREAD

**Issue by ali-masoudi** [qubvel/segmentation_models#157]
[Batch Normalization is Broken](http://blog.datumbox.com/the-batch-normalization-layer-of-keras-is-broken/)

read this post carefully and choose whether to apply it or not.

**Comment by qubvel** [comment]
Hi @ali-masoudi 
I know about this problem, that is why it is not recommended to freeze batchnorm in keras. And `encoder_freeze` param freeze layers in backbone except Batchnorms :smile:
